hyperspy/learn/mva.py

```diff
@@ -920,42 +919,217 @@ def get_explained_variance_ratio(self):
         s.axes_manager[-1].units = ''
         return s
 
-    def plot_explained_variance_ratio(self, n=50, log=True):
-        """Plot the decomposition explained variance ratio vs index number.
+    # def plot_explained_variance_ratio(self, n=50, log=True):
+    #     """Plot the decomposition explained variance ratio vs index number.
+    #
+    #     Parameters
+    #     ----------
+    #     n : int
+    #         Number of components.
+    #     log : bool
+    #         If True, the y axis uses a log scale.
+    #
+    #     Returns
+    #     -------
+    #     ax : matplotlib.axes
+    #
+    #     See Also:
+    #     ---------
+    #
+    #     `get_explained_variance_ration`, `decomposition`,
+    #     `get_decomposition_loadings`,
+    #     `get_decomposition_factors`.
+    #
+    #     """
+    #     s = self.get_explained_variance_ratio()
+    #     if n < s.axes_manager[-1].size:
+    #         s = s.isig[:n]
+    #     s.plot()
+    #     ax = s._plot.signal_plot.ax
+    #     # ax.plot(range(n), target.explained_variance_ratio[:n], 'o',
+    #     #         label=label)
+    #     ax.set_ylabel("Explained variance ratio")
+    #     ax.margins(0.05)
+    #     ax.autoscale()
+    #     ax.lines[0].set_marker("o")
+    #     ax.lines[0].set_linestyle("None")
+    #     if log is True:
+    #         ax.semilogy()
+    #     return ax
+
+    def plot_explained_variance_ratio(self, n=None, log=True, threshold=0,
+                                      hline=True, xaxis_type='index',
+                                      xaxis_labeling=None, signal_fmt=None,
+                                      noise_fmt=None, fig=None, ax=None,
+                                      **kwargs):
+        """Plot the decomposition explained variance ratio vs index number
+        (Scree Plot).
 
         Parameters
         ----------
-        n : int
-            Number of components.
+        n : int or None
+            Number of components to plot. If None, all components will be plot
         log : bool
             If True, the y axis uses a log scale.
+        threshold : float or int
```

Comment:
The functions doesn't behave as described here: it always plots the `hline` by default.  I personally like more the behaviour as described as it provides the information of whether the threshold was and int of perc visually. To implement it I bet that `hline` should default to "auto" instead of True.